### DISCUSSION TITLE
[V4] Get object's metadata from S3 with Adonis Driver

### DISCUSSION THREAD

**Discussion by Agilulfo1820** [adonisjs/core#2810] [Help] [answered]
Hello I'm using Adonis Driver to talk with AWS S3 (https://legacy.adonisjs.com/docs/4.1/file-system#_s3spaces_api).

Everything works fine but I am developing a job where I need to get the content type of all objects inside a folder. 
At the moment I'm doing 
```
const object = await Drive.disk('s3').getObject(filePath)
const type = object.ContentType
```
and it works.

The only thing is that this is not very performant, since it needs to get the full object only to get the metadata. I searched online and found out that AWS SDK provides `GetObjectMetadata()` method (https://stackoverflow.com/questions/10729754/how-can-i-get-the-mime-types-of-objects-in-amazon-s3). But to call that method I would need to implement AWS SDK. 

Is there a way I can do this with Driver('s3')?

**Comment by artem-a** [comment, marked as answer]
Hi @Agilulfo1820 why don't you use s3 instance? 
```javascript
const meta = await Drive.disk('s3').s3.headObject(params).promise()
```

[Docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#headObject-property)

I hope it helps

**Reply by Agilulfo1820** [reply]
Oh, so by doing `.s3` I can access all the AWS SDK? :o
